addon/components/link-to-component.js

```diff
@@ -15,6 +16,11 @@ export default LinkComponent.extend({
 
     let owner = getOwner(this);
 
+    assert(
+      `You attempted to use a link-to within a routeless engine, this is not supported.`,
```

Comment:
Do we have access to the Engine's `modulePrefix` so that we can let the user know which Engine this occurred in (ideally it'll be caught by the developer, but ya never know)? Also, we should probably mention something like:

`In order to construct links in a routeless, use {{link-to-external}}. See http://ember-engines.com/guide/linking-and-external-links for more info.`